### PR TITLE
fix: fix UX search after map moves

### PR DESCRIPTION
**Pull request by woyuen**
- Some assets don't appear after map moves
=> adjust max distance, depending on visible map bounds

- Search is triggered by map move but the checkbox button doesn't
appear
=> use an opt-out environment variable that enable search move
by default

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -34,6 +34,10 @@ VUE_APP_MAPBOX_TOKEN=
 VUE_APP_NOMINATIM_HOST=
 VUE_APP_NOMINATIM_KEY=
 
+# Automatically retrigger the search based on map center
+# Explicitly disable it by setting to false
+VUE_APP_SEARCH_AFTER_MAP_MOVE=
```
Comment (abruere) [inline .env.example]: Name does not convey it is an *opt-out*.
Something like `VUE_APP_DISABLE_AUTO_SEARCH_ON_MAP_MOVE` may be better.

Comment could also be more precise:

```
# A checkbox in UI lets users enable automatic search as they move the map.
# You can disable this feature by setting this variable to 'true'
```

File: src/store/mutation-types.js [hunk 2]
```diff
@@ -37,6 +37,10 @@ export const SET_DISPLAY_PRICE_RANGE = 'SET_DISPLAY_PRICE_RANGE'
 export const SET_SEARCH_DATES = 'SET_SEARCH_DATES'
 export const SET_SEARCH_LOCATION = 'SET_SEARCH_LOCATION'
 export const UNSET_SEARCH_LOCATION = 'UNSET_SEARCH_LOCATION'
+export const SET_MAP_CENTER = 'SET_MAP_CENTER'
+export const UNSET_MAP_CENTER = 'UNSET_MAP_CENTER'
+export const SET_MAX_DISTANCE = 'SET_MAX_DISTANCE'
+export const SEARCH__USE_MAP_CENTER = 'SEARCH__USE_MAP_CENTER'
```
Comment (abruere) [inline src/store/mutation-types.js]: You can merge two of these at least (see my comment in mutations file).

I would personally merge the four of them.

File: src/store/search/mutations.js [hunk 3]
```diff
@@ -40,6 +40,20 @@ export default {
...
     state.longitude = null
   },
+  [types.SET_MAP_CENTER] (state, { latitude, longitude }) {
+    state.mapCenterLatitude = latitude
+    state.mapCenterLongitude = longitude
+  },
+  [types.UNSET_MAP_CENTER] (state) {
+    state.mapCenterLatitude = null
+    state.mapCenterLongitude = null
+  },
```
Comment (abruere) [inline src/store/search/mutations.js]:
```

```
[types.SET_MAP_CENTER] (state, { latitude = null, longitude = null } = {}) {
    state.mapCenterLatitude = latitude
    state.mapCenterLongitude = longitude
  }
```

I think the two other mutations could also be merged with the above into a bigger `SEARCH__SET_MAP_OPTIONS` mutation.

File: src/pages/Search.vue [hunk 4]
```diff
@@ -274,46 +273,54 @@ export default {
...
-        })
-      })
+            this.populatedMapMarkers[assetId].style.display = 'none'
+          })
+          popup.on('close', () => {
+            // all popups (even empty ones) get closed when markers are destroyed
+            if (this.populatedMapMarkers[assetId]) this.populatedMapMarkers[assetId].style.display = ''
+          })
+        }
+      }, { concurrency: 2 }) // pMap
```
Comment (abruere) [inline src/pages/Search.vue]: Thanks for porting. I would mention it was ported from heroes-platform-demo, which might help debugging in the future.

File: .env.example (hunk 1 above)
Comment (woyuen) [reply .env.example]: Yes, thanks.

File: src/store/search/mutations.js (hunk 3 above)
Comment (woyuen) [reply src/store/search/mutations.js]: Right, I'll refactor it.

**Pull request merged**